sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -123,6 +126,24 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def hashCode(): Int = {
+        if (instance != null && instance.isInstanceOf[GenericUDFMacro]) {
```

Comment:
I mean when we check if this function wrapper is for a GenericUDFMacro, should we check the class name?